src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -253,9 +254,161 @@ IEnumerable<WorkloadPackId> ExpandPacks (WorkloadDefinitionId workloadId)
         /// <remarks>
         /// Used by the MSBuild workload resolver to emit actionable errors
         /// </remarks>
-        public IList<WorkloadInfo> GetWorkloadSuggestionForMissingPacks(IList<string> packId)
+        public ISet<WorkloadInfo> GetWorkloadSuggestionForMissingPacks(IList<string> packIds)
         {
-            throw new NotImplementedException();
+            var installedPacks = new HashSet<WorkloadPackId>();
+            var requestedPacks = new HashSet<WorkloadPackId>(packIds.Select(p => new WorkloadPackId(p)));
+
+            foreach (var pack in _packs)
+            {
+                var aliasedId = pack.Value.TryGetAliasForPlatformIds(_platformIds) ?? pack.Value.Id;
+                var packPath = GetPackPath(_dotNetRootPath, aliasedId, pack.Value.Version, pack.Value.Kind);
+
+                if (PackExists(packPath, pack.Value.Kind))
+                {
+                    installedPacks.Add(pack.Key);
+                }
+            }
+
+            // find workloads that contain any of the requested packs
+            var incompleteCandidates = new List<WorkloadSuggestionCandidate>();
+            var completeCandidates = new HashSet<WorkloadSuggestionCandidate>();
+
+            foreach (var workload in _workloads)
+            {
+                var expanded = new HashSet<WorkloadPackId>(ExpandWorkload(workload.Value));
+                if (expanded.Any(e => requestedPacks.Contains(e)))
+                {
+                    var stillMissing = new HashSet<WorkloadPackId>(requestedPacks.Where(p => !expanded.Contains(p)));
+                    var satisfied = requestedPacks.Count - stillMissing.Count;
+
+                    var candidate = new WorkloadSuggestionCandidate(new HashSet<WorkloadDefinitionId>() { workload.Key }, expanded, stillMissing);
+
+                    if (candidate.IsComplete)
+                    {
+                        completeCandidates.Add(candidate);
+                    }
+                    else
+                    {
+                        incompleteCandidates.Add(candidate);
+                    }
+                }
+            }
+
+            //find all valid complete permutations by recursively exploring possible branches from a root
```

Comment:
nit: same here.

And a method that worth comments every 3 lines is a dangerous method to edit later. We need more test around this method. And it is better to make it a separate method, even a separate class.